packages/@aws-cdk/aws-eks-next/lib/cluster.ts

```diff
@@ -29,47 +29,53 @@ const DEFAULT_CAPACITY_TYPE = ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.Inst
  * An EKS cluster
  */
 export interface ICluster extends IResource, ec2.IConnectable {
-  /**
-   * The VPC in which this Cluster was created
-   */
-  readonly vpc: ec2.IVpc;
 
   /**
    * The physical name of the Cluster
+   *
    * @attribute
    */
   readonly clusterName: string;
 
   /**
-   * The unique ARN assigned to the service by AWS
-   * in the form of arn:aws:eks:
+   * The unique ARN assigned to the service by AWS in the form of arn:aws:eks:
+   *
    * @attribute
    */
   readonly clusterArn: string;
 
+  /**
+   * The VPC in which this Cluster was created
+   */
+  readonly vpc?: ec2.IVpc;
+
   /**
    * The API Server endpoint URL
+   *
    * @attribute
    */
-  readonly clusterEndpoint: string;
+  readonly clusterEndpoint?: string;
 
   /**
    * The certificate-authority-data for your cluster.
+   *
    * @attribute
    */
-  readonly clusterCertificateAuthorityData: string;
+  readonly clusterCertificateAuthorityData?: string;
 
   /**
    * The cluster security group that was created by Amazon EKS for the cluster.
+   *
    * @attribute
    */
-  readonly clusterSecurityGroupId: string;
+  readonly clusterSecurityGroupId?: string;
```

Comment:
@eladb I changed this because this property is now checked in `KubectlProvider`. It made more sense to me to change this to optional rather than having to catch an exception that this will throw for imported clusters that don't specify this attribute. 

Since I changed this, i also changed all other optional attributes to be optional. Exceptions will be thrown by callers if needed and provide more contextual errors.

Reply:
It's against our guidelines. Let's discuss tomorrow

Reply:
Yes, I've scratched this PR in favor of: https://github.com/aws/aws-cdk/pull/10335

It contains minimal changes and doesn't turn everything to optional, I only later understood what are the implications of it. It does however create a quirk where we still allow passing `kubectlProviderSecurityGroup` even though its no longer strictly necessary.